pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/applicationconfiguration_without_traitdefinition_test.go

```diff
@@ -0,0 +1,232 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha2"
+	"github.com/oam-dev/kubevela/pkg/oam/util"
+)
+
+var _ = Describe("Test Deploying ApplicationConfiguration without TraitDefinition", func() {
+	const (
+		namespace     = "definition-test"
+		appName       = "hello"
+		componentName = "backend"
+	)
+	var (
+		ctx         = context.Background()
+		workload    v1alpha2.ContainerizedWorkload
+		component   v1alpha2.Component
+		workloadKey = client.ObjectKey{
+			Name:      componentName,
+			Namespace: namespace,
+		}
+		appConfig    v1alpha2.ApplicationConfiguration
+		appConfigKey = client.ObjectKey{
+			Name:      appName,
+			Namespace: namespace,
+		}
+		req = reconcile.Request{NamespacedName: appConfigKey}
+		ns  = corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+	)
+
+	BeforeEach(func() {})
+
+	It("ManualScalerTrait should work successfully even though its TraitDefinition doesn't exist", func() {
+		var componentStr = `
+apiVersion: core.oam.dev/v1alpha2
+kind: Component
+metadata:
+  name: backend
+  namespace: definition-test
+spec:
+  workload:
+    apiVersion: core.oam.dev/v1alpha2
+    kind: ContainerizedWorkload
+    spec:
+      containers:
+        - name: nginx
+          image: nginx:1.9.4
+          ports:
+            - containerPort: 80
+              name: nginx
+          env:
+            - name: TEST_ENV
+              value: test
+          command: [ "/bin/bash", "-c", "--" ]
+          args: [ "while true; do sleep 30; done;" ]
+`
+
+		var appConfigStr = `
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: hello
+  namespace: definition-test
+spec:
+  components:
+    - componentName: backend
+      traits:
+        - trait:
+            apiVersion: core.oam.dev/v1alpha2
+            kind: ManualScalerTrait
+            spec:
+              replicaCount: 2
+              workloadRef:
+                apiVersion: core.oam.dev/v1alpha2
+                kind: ContainerizedWorkload
+                name: backend
+`
+
+		By("Create namespace")
+		Eventually(
+			func() error {
+				return k8sClient.Create(ctx, &ns)
+			},
+			time.Second*3, time.Millisecond*300).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
+
+		By("Create Component")
+		Expect(yaml.Unmarshal([]byte(componentStr), &component)).Should(BeNil())
+		Expect(k8sClient.Create(ctx, &component)).Should(Succeed())
+		cmpV1 := &v1alpha2.Component{}
+		Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: componentName}, cmpV1)).Should(Succeed())
+
+		By("Create ApplicationConfiguration")
+		Expect(yaml.Unmarshal([]byte(appConfigStr), &appConfig)).Should(BeNil())
+		Expect(k8sClient.Create(ctx, &appConfig)).Should(Succeed())
+
+		By("Reconcile")
+		reconcileRetry(reconciler, req)
+		time.Sleep(5)
```

Comment:
Sleeping 5 seconds here can help unit-test on Github CI successfully pass, though it passed in my local unit-test.